src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php

```diff
@@ -246,7 +246,7 @@ private function compileRoute(Route $route, $name, $supportsRedirections, $paren
 EOF;
 
         if ($methods) {
-            $gotoname = 'not_'.preg_replace('/[^A-Za-z0-9_]/', '', $name);
+            $gotoname = 'not_'.preg_replace('/\W/', '', $name);
```

Comment:
I think the explicit version is more readable. Also we don't want locale specific things going on which the character classes are for. Otherwise it would be wrong, so just adding the unicode modifier would create a bug with your change. So I'm :-1: on all these regex changes.